File: sdks/php/src/sdk/src/Documents/Types/InitializeRenderJobRequestFormatOpts.php
<?php

namespace Pogodoc\Documents\Types;

use Pogodoc\Core\Json\JsonSerializableType;
use Pogodoc\Core\Json\JsonProperty;

class InitializeRenderJobRequestFormatOpts extends JsonSerializableType
{
    /**
     * @var ?float $fromPage
     */
    #[JsonProperty('fromPage')]
    public ?float $fromPage;

    /**
     * @var ?float $toPage
     */
    #[JsonProperty('toPage')]
    public ?float $toPage;

    /**
     * @var ?value-of<InitializeRenderJobRequestFormatOptsFormat> $format
     */
    #[JsonProperty('format')]
    public ?string $format;

    /**
     * @var ?string $waitForSelector
     */
    #[JsonProperty('waitForSelector')]
    public ?string $waitForSelector;

    /**
     * @param array{
     *   fromPage?: ?float,
     *   toPage?: ?float,
     *   format?: ?value-of<InitializeRenderJobRequestFormatOptsFormat>,
     *   waitForSelector?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->fromPage = $values['fromPage'] ?? null;
        $this->toPage = $values['toPage'] ?? null;
        $this->format = $values['format'] ?? null;
        $this->waitForSelector = $values['waitForSelector'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/sdk/src/PogodocApi/Documents/Types/InitializeRenderJobRequestFormatOpts.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using PogodocApi.Core;

namespace PogodocApi;

/// <summary>
/// Format options for the rendered document
/// </summary>
public record InitializeRenderJobRequestFormatOpts
{
    [JsonPropertyName("fromPage")]
    public double? FromPage { get; set; }

    [JsonPropertyName("toPage")]
    public double? ToPage { get; set; }

    [JsonPropertyName("format")]
    public InitializeRenderJobRequestFormatOptsFormat? Format { get; set; }

    /// <summary>
    /// Selector to wait for to know when the page is loaded and can be saved as pdf, png, etc.
    /// </summary>
    [JsonPropertyName("waitForSelector")]
    public string? WaitForSelector { get; set; }

    /// <summary>
    /// Additional properties received from the response, if any.
    /// </summary>
    /// <remarks>
    /// [EXPERIMENTAL] This API is experimental and may change in future releases.
    /// </remarks>
    [JsonExtensionData]
    public IDictionary<string, JsonElement> AdditionalProperties { get; internal set; } =
        new Dictionary<string, JsonElement>();

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
